src/Credit/Client.php

```diff
@@ -0,0 +1,181 @@
+
+        $response = $this->guzzleClient->request('post', $this->config->getUrl(), [
+            'login' => $this->config->getKey(),
+            'password' => $this->config->getSecret(),
+            'method' => static::BILL_CREATE,
+            'payeeId' => $this->config->getPayee(),
+            'contractNumber' => $creditToCard->getCardToken()
+        ]);
+
+        $this->validateTransfer($creditToCard, $response);
+
+        $document = simplexml_load_string(
+            $response->getBody()->__toString()
+        );
+
+        $response = $this->guzzleClient->request('post', $this->config->getUrl(), [
+            'login' => $this->config->getKey(),
+            'password' => $this->config->getSecret(),
+            'method' => static::BILL_PAY,
+            'billId' => $document->{static::TAG_BILL}->attributes()[static::ATTRIBUTE_ID],
+            'amount' => $creditToCard->getAmount()
+        ]);
+
+        $this->validateTransfer($creditToCard, $response);
+
+        $response = simplexml_load_string(
+            $response->getBody()->__toString()
+        )->{static::TAG_BILL};
+
+        return new Payment(
+            $response->attributes()[static::ATTRIBUTE_ID],
+            Carbon::createFromFormat('d.m.Y H:i:s', (string)$response->{static::TAG_PAY_DATE}),
+            (float)$response->{static::TAG_PAY_AMOUNT},
+            (string)$response->{static::TAG_STATUS}
+        );
+    }
+
+    /**
+     * @param Credit\TransferInterface $creditToCard
+     *
+     * @throws Credit\Exception
+     */
+    protected function validateCardNumber(Credit\TransferInterface $creditToCard): void
+    {
+        $card = $creditToCard->getCardToken();
+
+        if (!preg_match('/\d{16}/', $card)
+            && !$this->luhn->isValid(LuhnAlgorithm\Number::fromString($card))) {
+            throw new Credit\Exception($creditToCard, "Invalid card number");
+        }
+    }
+
+    /**
+     * @param string $payee
+     *
+     * @throws \Exception
+     */
+    protected function validatePayee(string $payee): void
+    {
+        if ($this->config->getPayee() !== $payee) {
+            throw new \Exception("Incorrect payee id returned from Portmone service");
+        }
+    }
+
+    /**
+     * @param Credit\TransferInterface $creditToCard
+     * @param ResponseInterface        $response
+     *
+     * @throws BillException
+     * @throws Credit\Exception
+     * @throws \Exception
+     */
+    protected function validateTransfer(
+        Credit\TransferInterface $creditToCard,
+        ResponseInterface $response
+    ): void {
+        $responseBody = $response->getBody()->__toString();
+
+        try {
+            $responseXml = simplexml_load_string($responseBody);
+
+            if ($responseXml === false) {
+                throw new \Exception("Invalid xml document.\nBODY: " . $responseBody);
+            }
+        } catch (\Throwable $exception) {
+            throw new Credit\Exception(
```

Comment:
Why do u use custom  exception here and native (\Exception) before?

Reply:
custom exception removed